### PR TITLE
step plots with plotly

### LINKED ISSUE #415: plotly timeseries plot does not use steps
I can't remember if we discussed this or not (I can't find anything), but it appears the plotly time series plots do not use steps for interval averages. 

It seems that we need a function like `line_or_step` for the plotly graphs.

https://github.com/SolarArbiter/solarforecastarbiter-core/blob/d1254d15bc0d4f4405bbab2ad9f195cbbca77332/solarforecastarbiter/plotting/utils.py#L45-L77

It [appears](https://plotly.com/python/line-charts/) plotly's `line_shape` kwarg is the way to go. 

Am I missing something @alorenzo175?

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #415 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [ ] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [ ] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: solarforecastarbiter/reports/figures/plotly_figures.py [hunk 1]
```diff
@@ -277,9 +286,11 @@ def timeseries(timeseries_value_df, timeseries_meta_df,
             name=_legend_text(metadata['forecast_name']),
             legendgroup=metadata['forecast_name'],
             marker=dict(color=next(palette)),
-            connectgaps=False),
+            connectgaps=False,
+            **plot_kwargs),
         )
         plotted_objects += 1
+    fig.update_traces(mode='lines+markers', marker=dict(size=6))
```
Comment (wholmgren) [inline solarforecastarbiter/reports/figures/plotly_figures.py:293]: comment out pending decision on whether or not its too busy

**Comment by wholmgren** [comment]
event:

newplot(7)

3 months with and without markers

newplot(8)
newplot(9)

File: solarforecastarbiter/plotting/utils.py [hunk 2]
```diff
@@ -77,3 +77,23 @@ def line_or_step(interval_label):
...
+    For a given interval_label, determine any kwargs for the plot.
+    """
+    if 'instant' in interval_label:
+        plot_kwargs = dict()
+    elif interval_label == 'beginning':
+        plot_kwargs = dict(line_shape='hv')
+    elif interval_label == 'ending':
+        plot_kwargs = dict(line_shape='vh')
+    elif interval_label == 'event':
+        plot_kwargs = dict(line_shape='vh')
```
Comment (alorenzo175) [inline solarforecastarbiter/plotting/utils.py:93]: To me 'hv' makes more sense for events. Basically, the idea is that once an event switches, you would expect it to continue at 1/0 until the next switch and the line after the point kinda signals that. :shrug:

File: solarforecastarbiter/reports/figures/plotly_figures.py (hunk 1 above)
Comment (alorenzo175) [reply solarforecastarbiter/reports/figures/plotly_figures.py:293]: yeah, it is too busy for most plots, but I like it for event plots. Up to you

File: solarforecastarbiter/plotting/utils.py (hunk 2 above)
Comment (wholmgren) [reply solarforecastarbiter/plotting/utils.py:93]: True. I just copied the bokeh setting, but will change both.

Screen Shot 2020-04-24 at 3 14 01 PM

**Pull request merged**